include/kernel.h

```diff
@@ -1532,7 +1625,79 @@ const char *k_thread_state_str(k_tid_t thread_id);
  *
  * @return Timeout delay value.
  */
-#define K_FOREVER (-1)
+#define K_FOREVER Z_FOREVER
+
+#ifdef CONFIG_TIMEOUT_64BIT
+
+/**
+ * @brief Generates an absolute/uptime timeout value in ticks
```

Comment:
shouldn't it be `from ticks` or `from absolute/uptime given in ticks`?

Reply:
Meh.  Grammatically: it's being documented as a constructor, not a conversion routine.  The timeout "is" the value in ticks.  But sure.